src/bulkQueries.ts

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs/promises';
+import * as path from 'path';
+import { Calculator, DRCalculationOutput } from 'fqm-execution';
+
+/**
+ * Loads file from cli input to calculate bulkQueries from the file
+ */
+export async function cliBulkQueries(filePath: string) {
+  // Read in bundle
+  let data: string;
+  try {
+    data = await fs.readFile(path.resolve(filePath), 'utf8');
+  } catch (err) {
+    console.error('Error reading the bundle: ', err);
+    return;
+  }
+
+  const bundle = JSON.parse(data) as fhir4.Bundle;
+  const queries = await bulkQueries(bundle);
+  console.log(JSON.stringify(queries));
+}
+
+/**
+ * Generates the queries that would come after a bulk "$export?"
+ */
+export async function bulkQueries(bundle: fhir4.Bundle) {
+  // simple approach...
+  // If there's a codeFilter with a direct code, create a typeFilter for that code
+  // If the codeFilter doesn't have a code (valueset or some other specification), create a generic _type query instead
+  // This _type query should supercede all other potential narrowing data requirements that are found on that resource
+  // If we can narrow:
+  // Within the same DR codeFilter list, typeFilters should be ANDed (i.e. _typeFilter=A,B)
+  // Across different DRs, typeFilters should be ORed (i.e. _typeFilter=A&_typeFilter=B)
+  // Make sure to also include a _type query for all resource types needed
+
+  const dataRequirements: DRCalculationOutput = await Calculator.calculateDataRequirements(bundle);
+
+  // create record resourcetype => [] of valid typeFilter query strings that will be separated by "&" and logically handled as ORs
+  const typeFilters: Record<string, string[] | undefined> = {};
+  dataRequirements.results.dataRequirement?.forEach(dr => {
+    //empty array is general _type query that overrides a more specific _typeFilter
+    if (typeFilters[dr.type]?.length === 0) return;
+
+    //any codeFilter that's non-coded or no-path or any contained codings have no code -> results in a general _type query
+    if (dr.codeFilter?.find(cf => !cf.code || !cf.path || cf.code.find(coding => !coding.code))) {
```

Comment:
```suggestion
    if (dr.codeFilter?.some(cf => !cf.code || !cf.path || cf.code.some(coding => !coding.code))) {
```
Use some to return a boolean since thats whats needed for the predicate and the if statement.